src/borrower/feed/navfeed.sol

```diff
@@ -219,15 +221,15 @@ abstract contract NAVFeed is Auth, Discounting {
         loanDetails[loan].borrowed = toUint128(safeAdd(borrowed(loan), amount));
 
         // return increase NAV amount
-        return calcDiscount(discountRate.value, fv, uniqueDayTimestamp(block.timestamp), maturityDate_);
+        return calcDiscount(discountRate.value, fv, nnow, maturityDate_);
     }
 
     function repay(uint loan, uint amount) external virtual auth {
         calcUpdateNAV();
 
         // In case of successful repayment the latestNAV is decreased by the repaid amount
         bytes32 nftID_ = nftID(loan);
-        uint maturityDate_ = maturityDate(nftID_);
+        uint maturityDate_ = uniqueDayTimestamp(maturityDate(nftID_));
```

Comment:
The maturity date gets the `uniqueDayTimestamp` in the file method, so it is not necessary here.